### PR TITLE
1345: context: register stack with valgrind

### LINKED ISSUE #1345: Register ULT stacks with valgrind if headers are present while compiling
**What Needs to be Done?**

Call `VALGRIND_STACK_REGISTER`, etc.

### PR DESCRIPTION
**Pull request by lifflander**
Fixes #1345

### CODE REVIEW

**Comment by PhilMiller** [comment]
Code looks fine. Confirm that at least one of the CI builds has valgrind installed with the headers, and so is testing this code?

**Comment by PhilMiller** [comment]
And maybe add some separate test to the script that runs something simple under valgrind.

For bonus points, write a "must fail" test to run under valgrind and affirm that it detects an error.

**Comment by lifflander** [comment]
> And maybe add some separate test to the script that runs something simple under valgrind.
> 
> For bonus points, write a "must fail" test to run under valgrind and affirm that it detects an error.

I could even make the macro call to make sure it exists in the same form?

**Comment by lifflander** [comment]
@cz4rs @PhilMiller @JacobDomagala @jstrzebonski Could I get a review on this PR?

**Review by PhilMiller — APPROVED** [review]
It's still not clear that this is getting tested, but the code looks exactly as expected.

File: lib/context/check_valgrind.cmake [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  valgrind_stack_register.cmake.cc
+)
+
+if (context_valgrind_stack_compiled)
+  message(
+    STATUS "Successfully compiled with valgrind stack registration"
+  )
+else()
+  message(
+    STATUS "Failed to compile with valgrind stack registration"
```
Comment (JacobDomagala) [inline lib/context/check_valgrind.cmake:40]: Should we compile vt even if this fails? I think we could use `FATAL_ERROR` instead of `STATUS` here

**Review by JacobDomagala — APPROVED** [review]

File: lib/context/check_valgrind.cmake (hunk 1 above)
Comment (PhilMiller) [reply lib/context/check_valgrind.cmake:40]: We definitely don't want to require that valgrind be installed to build vt. It's purely a debugging tool

Comment (lifflander) [reply lib/context/check_valgrind.cmake:40]: Agreed

**Pull request merged**